src/main/java/com/devonfw/sample/archunit/general/violations/ThirdPartyObjectOK.java

```diff
@@ -0,0 +1,9 @@
+package com.devonfw.sample.archunit.general.violations;
+
+// compliant: standard java object method equals()
+public class ThirdPartyObjectOK {
+
+    Integer one = 1;
+    boolean result = one.equals((Integer) 1);
```

Comment:
Actually the rule is about the null-safe equals `java.util.Objects.equals(first, second)` rather than the build in `Object.equals` method.